tutorial/markdown/aspnet/semantic-kernel/semantic-kernel-tutorial.md

```diff
@@ -0,0 +1,387 @@
+short_title: Vector Search with Semantic Kernel
+description:
+  - Build a semantic search application using Couchbase BHIVE vector index with Semantic Kernel.
+  - Learn to use the Couchbase .NET Vector Store Connector for Microsoft Semantic Kernel.
+  - Discover how to generate embeddings with OpenAI and store them in Couchbase.
+  - Perform vector similarity searches with filtering using SQL++ and ANN_DISTANCE.
+content_type: tutorial
+filter: sdk
+technology:
+  - fts
+  - kv
+tags:
+  - Semantic Kernel
+  - OpenAI
+  - Artificial Intelligence
+sdk_language:
+  - csharp
+length: 30 Mins
+---
+
+## Repository Links
+
+- **Connector Repository**: [couchbase-semantic-kernel](https://github.com/Couchbase-Ecosystem/couchbase-semantic-kernel) - The official Couchbase .NET Vector Store Connector for Microsoft Semantic Kernel
+- **This Example**: [CouchbaseVectorSearchDemo](https://github.com/Couchbase-Ecosystem/couchbase-semantic-kernel/tree/Support-Bhive-and-Composite-Index/CouchbaseVectorSearchDemo) - Complete working example demonstrating vector search with Couchbase
+
+## Introduction
+
+This demo showcases the **Semantic Kernel Couchbase connector** - a .NET library that bridges Microsoft's Semantic Kernel framework with Couchbase's vector search capabilities. The connector provides a seamless integration that allows developers to build AI-powered applications using familiar Semantic Kernel abstractions while leveraging Couchbase's vector indexing for high-performance semantic search.
+
+The connector supports three index types:
+- **BHIVE** (Hyperscale Vector Index) - for pure vector search at scale ← *Used in this demo*
+- **Composite Vector Index** - for vector search with heavy scalar filtering
+- **FTS** (Full-Text Search) - for hybrid text + semantic search
+
+This makes the connector ideal for RAG (Retrieval-Augmented Generation) applications, semantic search engines, hybrid search, and recommendation systems.
+
+## Prerequisites
+
+### 1. Couchbase Server Setup
+- **Couchbase Server 8.0+**
+- Local installation or Couchbase Cloud/Capella
+- Bucket with proper read/write permissions
+- Query service enabled for SQL++ operations
+
+### 2. OpenAI API Access
+- **OpenAI API Key** - Get one from: https://platform.openai.com/api-keys
+- Used for generating text embeddings with `text-embedding-ada-002` model
+- Ensure you have sufficient API quota for embedding generation
+
+### 3. Development Environment
+- **.NET 8.0** or later
+- Visual Studio, VS Code, or JetBrains Rider
+- Basic understanding of C# and vector databases
+
+
+## Setting Up the Environment
+
+### 1. Clone and Navigate
+```bash
+git clone https://github.com/Couchbase-Ecosystem/couchbase-semantic-kernel.git
+cd couchbase-semantic-kernel/CouchbaseVectorSearchDemo
+```
+
+### 2. Install Dependencies
+```bash
+dotnet restore
+```
+
+### 3. Configuration Setup
+
+Update `appsettings.Development.json` with your credentials:
```

Comment:
Should the bucket, scope & collection exist?

Reply:
You can also mention that these values can be changed with corresponding updation of the code.